cmd/tink-worker/client/client_test.go

```diff
@@ -0,0 +1,222 @@
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.GetWorkflowDataResponse{}, nil
+}
+
+func (s *fakeServer) GetWorkflowContext(context.Context, *workflow.GetRequest) (*workflow.WorkflowContext, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.WorkflowContext{}, nil
+}
+
+func (s *fakeServer) GetWorkflowContexts(*workflow.WorkflowContextRequest, workflow.WorkflowService_GetWorkflowContextsServer) error {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return nil
+}
+
+func (s *fakeServer) GetWorkflowContextList(context.Context, *workflow.WorkflowContextRequest) (*workflow.WorkflowContextList, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.WorkflowContextList{}, nil
+}
+
+func (s *fakeServer) ListWorkflows(*workflow.Empty, workflow.WorkflowService_ListWorkflowsServer) error {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return nil
+}
+
+func (s *fakeServer) ShowWorkflowEvents(*workflow.GetRequest, workflow.WorkflowService_ShowWorkflowEventsServer) error {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return nil
+}
+
+func (s *fakeServer) UpdateWorkflowData(context.Context, *workflow.UpdateWorkflowDataRequest) (*workflow.Empty, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.Empty{}, nil
+}
+
+func (s *fakeServer) ReportActionStatus(context.Context, *workflow.WorkflowActionStatus) (*workflow.Empty, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.Empty{}, nil
+}
+
+const bufSize = 1024 * 1024
+
+func startWorkflowServerAndConnectClient(t *testing.T, name string, server *grpc.Server) workflow.WorkflowServiceClient {
+	t.Helper()
+
+	ctx := context.Background()
+	listener := bufconn.Listen(bufSize)
+	go func() {
+		t.Helper()
+
+		if err := server.Server().Serve(listener); err != nil {
+			t.Error(fmt.Errorf("%s.Serve exited with error: %w", name, err))
+		}
+	}()
+
+	dialer := func(ctx context.Context, _ string) (net.Conn, error) {
+		return listener.DialContext(ctx)
+	}
+
+	conn, err := ggrpc.DialContext(ctx, "bufnet", ggrpc.WithContextDialer(dialer), ggrpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial %s: %v", name, err)
+	}
+
+	client := workflow.NewWorkflowServiceClient(conn)
+	return client
+}
+
+func TestObtainTinkServerCreds(t *testing.T) {
+	t.Run("certURL arg validation", func(t *testing.T) {
+		expectedErrorMsg := "certURL cannot be empty"
+		_, err := ObtainTinkServerCreds("")
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
```

Comment:
Another general comment: I see there are several calls to `t.Errorf`... Should these be `t.Fatalf`? The difference being, `Errorf` allows the test to keep going, and `Fatalf` stops the test right there.

Reply:
I'm pretty sure in all of these cases, there is no further code that's being run after the `t.Error()`, e.g, in the above snippet we can only run one or the other. What I wasn't clear on is whether calling `Fatal()` would prevent it from running any other tests? If you can confirm that's not he case, I can do a mass search/replace to change all of those Error's to Fatal's.